internal/engine/dcwatch/disable_watcher.go

```diff
@@ -0,0 +1,85 @@
+package dcwatch
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/tilt-dev/tilt/pkg/logger"
+
+	"github.com/tilt-dev/tilt/internal/dockercompose"
+	"github.com/tilt-dev/tilt/internal/store"
+	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
+	"github.com/tilt-dev/tilt/pkg/model"
+)
+
+type DisableSubscriber struct {
+	dcc dockercompose.DockerComposeClient
+}
+
+// How does the `New` function relate to the subscriber we're creating?
+func NewDisableSubscriber(dcc dockercompose.DockerComposeClient) *DisableSubscriber {
+	return &DisableSubscriber{
+		dcc: dcc,
+	}
+}
+
+func (w *DisableSubscriber) OnChange(ctx context.Context, st store.RStore, summary store.ChangeSummary) error {
+	if summary.IsLogOnly() {
+		return nil
+	}
+
+	state := st.RLockState()
+	project := state.DockerComposeProject()
+
+	if model.IsEmptyDockerComposeProject(project) {
+		st.RUnlockState()
+		return nil
+	}
+
+	var specsToDisable []model.DockerComposeUpSpec
+	for _, uir := range state.UIResources {
```

Comment:
Since `UIResources` are the source-of-truth, you can:
 * Short-circuit (instead of log only) if `len(summary.UIResources.Changes) == 0`
 * Iterate over `summary.UIResources.Changes` and then look up key in `state.UIResources` to only look at modified `UIResource`s

Reply:
neat!

Reply:
oops, turns out nothing was actually populating that field